docs/DeveloperGuide.adoc

```diff
@@ -358,13 +397,221 @@ image::UndoRedoActivityDiagram.png[width="650"]
+and displays to the feedback to the user by returning a `CommandResult` with the found data as argument.
+
+The following activity diagram summarizes the execution of the `balance` command.
+
+image::BalanceCommandActivityDiagram.png[width="800"]
+
+==== Design Considerations
+
+===== Aspect: Making balance not an undoable or re-doable command.
+
+* **Alternative 1 (current choice)**: Make `balance` not an undoable command**.
+
+* **Pros**: There was no need to make balance a re-doable or undoable command since it serves a purpose similar to `list`:
+it only displays information found in TravelBanker, but does not alter it. Thus, the user will not have any issues with this
+command should he make a mistake: it does not write any data.
+
+* ** Cons**: No significant disadvantage.
+
+* **Alternative 2: make `balance` an undoable command.**
+* **Pros**: No serious advantage, as explained in Alternative 1. Since the command does not aim to modify data,
+this method is not applicable.
+* **Cons**:  Extra layer of complexity that does not give any significant benefit
+
+===== Aspect: How balance command result is displayed.
+* **Alternative 1 (current choice)** : Pass the result as a string in the feedback to user.
+* **Pros** : Easy to implement and serves the purpose perfectly.
+* **Cons** : Value cannot be singled out to apply UI effects, such as colors.
+
+* **Alternative 2 ** : Single out result to make it modifiable by UI.
+* **Pros**: Allow for more pleasant user experience.
+* **Cons**: More difficult to implement and not much value added.
+// end::balancefeature[]
+
+// tag::sortfeature[]
+=== sort Feature
+The sort feature is implemented by the `sortCommand` class, which resides in the `Logic` component.
+When `sortCommand` is executed, it would call the `Model` component (`UniquePersonList`) to sort the person list. To specify the sorting order and the keyword to be sorted, two strings would be passed as parameters.
+In support of different keyword, Class `Name`, `Phone`, `Email`, `Address`, `Money` were add with a `compareTo` method for the creation of comparators.
+
+The sortCommand supports sorting of the filteredList (i.e. list that comes from FindCommand execution).
+
+In the current implementation, the sorted result would not be store in the storage.
+
+Commands that cannot be undone such as sortCommand are implemented like this:
+
+[source,java]
+----
+public class sortCommand extends Command {
+    @Override
+    public CommandResult execute() {
+        // ... sort logic ...
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+}
+----
+
+The following activity diagram summarizes the execution of the `sort` command.
+
+image::SortCommand_Activity_Diagram.png[width="800"]
+
+// end::sortfeature[]
+
 // tag::dataencryption[]
 === [Proposed] Data Encryption
 
 _{Explain here how the data encryption feature will be implemented}_
 
 // end::dataencryption[]
 
+// tag::moneyfield[]
+=== Add/Edit Money Field
```

Comment:
I think you can better describe these using diagrams, instead of explaining the changes in each component.